src/common/pf_lldp.c

```diff
@@ -440,10 +533,239 @@ void pf_lldp_send(
+         }
+         else
+         {
+            LOG_ERROR(PF_LLDP_LOG, "LLDP(%d): Port Id too long (%zu)\n", __LINE__, frame.len - 1);
+         }
+         break;
+      case LLDP_TYPE_TTL:
+         lldp_peer_info->ttl = pf_get_uint16(&parse_info, &offset);
+         break;
+      case LLDP_TYPE_ORG_SPEC:
+         {
+            uint8_t org_id[3];
+            uint8_t org_subtype;
+            pf_get_mem(&parse_info, &offset, sizeof(org_id), org_id);
+            org_subtype = pf_get_byte(&parse_info, &offset);
+
+            if (memcmp(org_id_pnio, org_id, sizeof(org_id)) == 0)
+            {
+               switch (org_subtype)
+               {
+               case LLDP_PNIO_SUBTYPE_MEAS_DELAY_VALUES:
+                  lldp_peer_info->port_delay.rx_delay_local    = pf_get_uint32(&parse_info, &offset);
+                  lldp_peer_info->port_delay.rx_delay_remote   = pf_get_uint32(&parse_info, &offset);
+                  lldp_peer_info->port_delay.tx_delay_local    = pf_get_uint32(&parse_info, &offset);
+                  lldp_peer_info->port_delay.tx_delay_remote   = pf_get_uint32(&parse_info, &offset);
+                  lldp_peer_info->port_delay.cable_delay_local = pf_get_uint32(&parse_info, &offset);
+                  break;
+               case LLDP_PNIO_SUBTYPE_PORT_STATUS:
+                  pf_get_mem(&parse_info, &offset, 4, lldp_peer_info->port_status);
+                  break;
+               case LLDP_PNIO_SUBTYPE_CHASSIS_MAC:
+                  pf_get_mem(&parse_info, &offset, 6, &lldp_peer_info->mac_address.addr);
+                  break;
+               default:
+                  offset += (frame.len - 4);
+                  break;
+               }
+            }
+            else if (memcmp(org_id_ieee_8023, org_id, sizeof(org_id)) == 0)
+            {
+               switch (org_subtype)
+               {
+               case LLDP_IEEE_SUBTYPE_MAC_PHY:
+                  lldp_peer_info->phy_config.aneg_support_status   = pf_get_byte(&parse_info, &offset);
+                  lldp_peer_info->phy_config.aneg_cap              = pf_get_uint16(&parse_info, &offset);
+                  lldp_peer_info->phy_config.operational_mau_type  = pf_get_uint16(&parse_info, &offset);
+                  break;
+               default:
+                  offset += (frame.len - 4);
+                  break;
+               }
+            }
+         }
+         break;
+      default:
+         offset += frame.len;
+         break;
+      }
+      frame = pf_get_lldp_tlv(&parse_info, &offset);
+   }
+
+   if (parse_info.result != PF_PARSE_OK)
+   {
+      LOG_ERROR(PF_LLDP_LOG, "LLDP(%d): Parsing error\n", __LINE__);
+      return -1;
+   }
+   return 0;
+}
+
+/**
+ * @internal
+ * Generate an alias name from port_id and chassis_id
+ *
+ * TODO - It is uncertain if the alias is name generation
+ *        should depend on the port and chassis id sub types.
+ *        Current implementation may work but uncertain why.
+ *
```

Comment:
Defined in Profinet 2.4, protocol section 4.3.1.4.18

Reply:
Ok